pageserver/src/tenant.rs

```diff
@@ -3312,6 +3320,18 @@ impl Tenant {
                 ancestor_children.push((timeline_entry.get_ancestor_lsn(), *timeline_id));
             }
         });
+        timelines_offloaded
```

Comment:
nit: would be nice to avoid the duplication with the `for_each` call above. You could map both iterators to TimelineOrOffloaded, chain the iterators and have one iteration.

Reply:
I tried doing it but then I realized that `TimelineOrOffloaded` holds an `Arc`, so we'd have to clone Arcs in order to do it, which has a non-zero overhead. I think it's okay to have a small amount of duplication instead.